### PR TITLE
test_maintenance_mode: fix flakiness

### PR DESCRIPTION
**Pull request by kbr-scylla**
Wait until CQL is available and nodes see each other before trying to perform a query.

### CODE REVIEW

**Comment by kbr-scylla** [comment]
Similar to https://github.com/scylladb/scylladb/pull/17040

File: test/topology_custom/test_maintenance_mode.py [hunk 1]
```diff
@@ -88,6 +91,8 @@ async def test_maintenance_mode(manager: ManagerClient):
     # Restart in normal mode to see if the changes made in maintenance mode are persisted
     await manager.server_update_config(server_a.server_id, "maintenance_mode", "false")
     await manager.server_start(server_a.server_id, wait_others=1)
+    await wait_for_cql_and_get_hosts(cql, [server_a], time.time() + 60)
```
Comment (denesb) [inline test/topology_custom/test_maintenance_mode.py:94]: Maybe we need to mention in `server_start()` (`manager_client.py`) that `server_start()` will only wait for the node to finish initializing and start listening on the CQL port, it does *not* wait for the other nodes to see it as UP.

Although, I imagine most people copy-paste these snippets from some other test and they won't see the comment. Still, it wouldn't hurt.

**Review by denesb — APPROVED** [review]

File: test/topology_custom/test_maintenance_mode.py (hunk 1 above)
Comment (kbr-scylla) [reply test/topology_custom/test_maintenance_mode.py:94]: Alternative: dtest has `wait_other_notice` parameter in their version of `server_start`, if `True` the function will wait for other servers to notice the restarted server as UP.

That would get rid of the need for explicit `servers_see_each_other`

And we could do similar for the CQL wait; we could have a `wait_for_cql` parameter, which would get a `Session | None` object. The default would be `get_cql()` (from within `ManagerClient`), but test could also pass `None` (if it cannot have a session at this point for some reason) or some other session if it needs to use one.

Comment (denesb) [reply test/topology_custom/test_maintenance_mode.py:94]: Even better, the less boilerplate, the better (we could leave the current exploded methods too, for a test which needs more fine-grained control).

**Comment by raphaelsc** [comment]
looks like it still fails after this patch:

`WARN  2024-02-02 14:14:46,881 [shard 0:stmt] gossip - Fail to apply application_state: std::runtime_error (endpoint_state_map does not contain endpoint = 127.51.71.8, application_states = {{RPC_READY -> Value(1,1)}})`

but maybe unrelated to gossip

```
            if resp.status != 200:
                text = await resp.text()
>               raise HTTPError(uri, resp.status, params, json, text)
E               test.pylib.rest_client.HTTPError: HTTP error 500, uri: http+unix://api/cluster/server/111/start, params: None, json: {'expected_error': None}, body:
E               getaddrinfo() argument 1 must be string or None
```

**Comment by kbr-scylla** [comment]
@raphaelsc update your driver and submodules (cqlsh)

**Pull request closed without merging**